### PR TITLE
Remove enzyme from tests

### PR DESCRIPTION
**Pull request by lucapircher**
# Description

Removes the use of enzyme for tests to unblock us from upgrading to React 18 and Nextjs 13.

- refactors tests to use react testing library
- removes enzyme
- removes a few tests that do not provide a lot of value

### General

- [ ] Code Review
- [x] Test Coverage is equal or higher
- [x] Update ticket on the TODO board
- [x] (if .env file changes) Notify other team members

### CODE REVIEW

File: services/site/tests/components/challenge/Editors.spec.tsx [hunk 1]
```diff
@@ -25,21 +25,23 @@ const handleReset = jest.fn();
...
 
-    expect(wrapper.exists("div")).toBeTruthy();
-    expect(wrapper.find("div").first().props().className).toContain(mockClassName);
+    // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+    expect(container.querySelector(`div.${mockClassName}`)).toBeInTheDocument();
   });
 
-  it("renders one editor", () => {
-    const wrapper = shallow(<Editors className={mockClassName} editors={[htmlEditorConfig]} onReset={handleReset} />);
+  it("can render a single editor", () => {
```
Comment (dnikub) [inline services/site/tests/components/challenge/Editors.spec.tsx]: I would maybe keep the wording for tests in this file at "renders XYZ", as all other tests in the project start that way.

File: services/site/tests/components/challenge/Sidebar.spec.tsx [hunk 2]
```diff
@@ -26,40 +24,33 @@ const mockLevel: Pick<CodeLevel, "instructions" | "tasks"> = {
 describe("Sidebar", () => {
   it("renders the wrapper element with classes", () => {
     const mockClassName = "mock-class";
-    const wrapper = shallow(<Sidebar className={mockClassName} challengeName={mockChallengeName} level={mockLevel} />);
+    const { container } = render(<Sidebar className={mockClassName} challengeName={mockChallengeName} level={mockLevel} />);
 
-    expect(wrapper.find("section").props().className).toContain(mockClassName);
-  });
-
-  it("renders the scroll overlay", () => {
```
Comment (dnikub) [inline services/site/tests/components/challenge/Sidebar.spec.tsx:34]: Why did you remove this test?

File: services/site/tests/components/challenge/level/CodeLevel.spec.tsx [hunk 3]
```diff
@@ -95,40 +89,38 @@ describe("Code Level", () => {
...
+    }));
   });
 
   afterEach(() => {
     jest.resetAllMocks();
   });
 
-  it("renders `Sidebar` component", () => {
-    const view = mountCodeLevel();
+  it("renders instructions", () => {
```
Comment (dnikub) [inline services/site/tests/components/challenge/level/CodeLevel.spec.tsx]: I would add "(Sidebar)" to the test description so it is more understandable which instructions are meant.

File: services/site/tests/components/challenge/level/QuizLevel.spec.tsx [hunk 4]
```diff
@@ -70,79 +70,101 @@ const mocks: MockedResponse[] = [
...
-    expect(wrapper.exists(SingleAnswer)).toBeTruthy();
+    expect(screen.getByText(mockAnswers.at(0).text)).toBeInTheDocument();
   });
 
-  it("renders fail animation when quiz status is fail", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
+  it("renders fail animation when quiz status is fail", async () => {
+    render(
+      <MockedProvider mocks={mockedResponsesForSuccess}>
```
Comment (dnikub) [inline services/site/tests/components/challenge/level/QuizLevel.spec.tsx]: Why is it the mock for success here, when the test checks if there is the FAIL animation?

File: services/site/tests/components/challenge/level/QuizLevel.spec.tsx [hunk 5]
```diff
@@ -70,79 +70,101 @@ const mocks: MockedResponse[] = [
...
         <QuizLevel question={mockText} answers={mockAnswers} isLastLevel={false} levelId={"1"} />
       </MockedProvider>,
     );
 
-    expect(wrapper.find(Lottie).length).toBe(0);
```
Comment (dnikub) [inline services/site/tests/components/challenge/level/QuizLevel.spec.tsx:128]: maybe I just don't understand it correctly, but how do you check if the Lottie animation is present now?

File: services/site/tests/components/challenge/level/QuizLevel.spec.tsx [hunk 6]
```diff
@@ -70,79 +70,101 @@ const mocks: MockedResponse[] = [
...
+
+    expect(screen.getByTestId("lottie")).toBeInTheDocument();
   });
 
-  it("renders success animation when quiz status is success", () => {
-    const wrapper = mount(
-      <MockedProvider mocks={mocks}>
+  it("renders success animation when quiz status is success", async () => {
+    render(
+      <MockedProvider mocks={mockedResponsesForIncorrect}>
```
Comment (dnikub) [inline services/site/tests/components/challenge/level/QuizLevel.spec.tsx]: Why is it the mock for fail here, when the test checks if there is the SUCCESS animation?

File: services/site/tests/components/challenge/sidebar/HintBox.spec.tsx [hunk 7]
```diff
@@ -16,45 +15,40 @@ const hints = [
 ];
 
 describe("HintBox", () => {
-  it("renders closed hint box", () => {
-    const wrapper = mount(<HintBox hints={hints} />);
+  it("renders the box closed by default", () => {
```
Comment (dnikub) [inline services/site/tests/components/challenge/sidebar/HintBox.spec.tsx]: I would write "..the HintBox closed..." so it is in alignment with the naming of all other tests in this file.

File: services/site/tests/components/homepage/ChallengeHeader.spec.tsx [hunk 8]
```diff
@@ -46,16 +41,4 @@ describe("ChallengeHeader", () => {
     expect(userAccountModalApi.show).toHaveBeenCalledTimes(1);
     expect(userAccountModalApi.show).toHaveBeenCalledWith("signup");
   });
-
-  it("renders the github button", () => {
```
Comment (dnikub) [inline services/site/tests/components/homepage/ChallengeHeader.spec.tsx:50]: why did you remove these two tests?

File: services/site/tests/components/challenge/Sidebar.spec.tsx (hunk 2 above)
Comment (lucapircher) [reply services/site/tests/components/challenge/Sidebar.spec.tsx:34]: If converted to testing library this convert to checking if a div is present, which by itself does not provide any value. The overlay is tested by itself so we are not missing testing any behaviour without this test.

File: services/site/tests/components/challenge/level/QuizLevel.spec.tsx (hunk 4 above)
Comment (lucapircher) [reply services/site/tests/components/challenge/level/QuizLevel.spec.tsx]: Ups. Mixed those two up...

File: services/site/tests/components/challenge/level/QuizLevel.spec.tsx (hunk 5 above)
Comment (lucapircher) [reply services/site/tests/components/challenge/level/QuizLevel.spec.tsx:128]: At the top of the file we define a mock for the Lottie component that is rendered instead of the actual Lottie animation. That component contains a `data-testid="lottie"` attribute. If we can find an element with that test id (done via `.getByTestId("lottie")`, we can assume that the animation will be rendered in the actual application (instead of the mock).
I hope that somewhat explains it. If not, let me know.

File: services/site/tests/components/challenge/level/QuizLevel.spec.tsx (hunk 6 above)
Comment (lucapircher) [reply services/site/tests/components/challenge/level/QuizLevel.spec.tsx]: That's the second part of the mixup...

File: services/site/tests/components/homepage/ChallengeHeader.spec.tsx (hunk 8 above)
Comment (lucapircher) [reply services/site/tests/components/homepage/ChallengeHeader.spec.tsx:50]: Testing if the buttons are present do not provide any additional value to checking if the sign up button is rendered if the user is logged in.
In my opinion we would have to test some kind of behaviour here to justify having additional tests.

**Review by dnikub — APPROVED** [review]
